### DISCUSSION TITLE
Metadata could not be deserialized

### DISCUSSION THREAD

**Discussion by musi00** [International-Data-Spaces-Association/DataspaceConnector#62] [Q&A] [answered]
Hello,

I have a provider connector up and running and I'm trying to get the description of the resource via the REST API: `/admin/api/request/description`.  However this fails with the error: `"Metadata could not be deserialized"`.  I can successfully get the description of the connector (i.e when I leave the resource-id field blank) but it fails when I use the resource-id.
I can also get the resource description successfully via the Resrouce Handling API:  `/admin/api/resources/{resource-id}`.  

Here is a screenshot of the Failed API Request:

![Screenshot_1](https://user-images.githubusercontent.com/5386051/109594006-08924a00-7ae0-11eb-8975-eaa85c914f99.png)

I actually have 2 connectors setup (provider and consumer) and consumer has to be able to get the description of the resource in order to negotiate the contract as discussed in the wiki.  However I get the same error when I do it from the consumer side.  The above error happens from the provider side (i.e the connector is getting the description of its own resource, this was done for debugging).

**Comment by juliapampus** [comment, marked as answer]
Hi! The id you are requesting is wrong. Within the connector's self-description, each resource has an id like that: `"@id" : "https://w3id.org/idsa/autogen/resource/0573bfc3-4627-4c73-a0db-54745e120485",`. You have to request this one.

**Reply by musi00** [reply]
Hi Julia, thank you for your reply.  I corrected the id but I still get the same error:

![image](https://user-images.githubusercontent.com/5386051/109651953-2a65ee00-7b2d-11eb-89eb-78564896c1cb.png)

**Reply by juliapampus** [reply]
What version do you use?

**Reply by musi00** [reply]
v4.0.2

**Reply by musi00** [reply]
I tried again with v4.1.0 and this time it worked.  Also when requesting the artifact from the consumer side, the **representation id** must be used to negotiate the contract and then this same representation id must be used to request the artifact after the transfer contract URI is provided.  I was trying to use the resource id initially and that didn't work.  Just in case someone else ran into a similar issue.
